commit d82978ca1593890a1b41eab6d06fe6e5950e4722
Author: Kent Overstreet <kent.overstreet@linux.dev>
Date:   Wed Jul 12 11:43:03 2023 -0400

    bcachefs: Add a race_fault() for write buffer slowpath
    
    We haven't hooked up dynamic fault injection quite yet, but we will soon
    
    Signed-off-by: Kent Overstreet <kent.overstreet@linux.dev>

diff --git a/fs/bcachefs/btree_write_buffer.c b/fs/bcachefs/btree_write_buffer.c
index b50226313a47..6c30a72e6eee 100644
--- a/fs/bcachefs/btree_write_buffer.c
+++ b/fs/bcachefs/btree_write_buffer.c
@@ -129,6 +129,9 @@ int __bch2_btree_write_buffer_flush(struct btree_trans *trans, unsigned commit_f
 	keys = wb->keys[s.idx];
 	nr = s.nr;
 
+	if (race_fault())
+		goto slowpath;
+
 	/*
 	 * We first sort so that we can detect and skip redundant updates, and
 	 * then we attempt to flush in sorted btree order, as this is most